### PR TITLE
Fix parsing of huge OpenQASM 2 conditionals

### LINKED ISSUE #12773: Integer overflow when creating circuits from QASM files with large ints
### Environment

- **Qiskit version**:  1.1.1
- **Python version**: 
- **Operating system**: 

### What is happening?

As reported https://github.com/Qiskit/benchpress/issues/17, qasm imports lead to int overflows when qasm files contain large ints.  E.g. this example will fail

```
OPENQASM 2.0;
include "qelib1.inc";
qreg q0[301];
creg c0[301];
h q0[300];
measure q0[300] -> c0[300];
if(c0==2037035976334486086268445688409378161051468393665936250636140449354381299763336706183397376) h q0[0];
```

with 

```
thread '<unnamed>' panicked at crates/qasm2/src/lex.rs:279:42:
called `Result::unwrap()` on an `Err` value: ParseIntError { kind: PosOverflow }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
PanicException: called `Result::unwrap()` on an `Err` value: ParseIntError { kind: PosOverflow }
```

### How can we reproduce the issue?

Run the above QASM file with `QuantumCircuit.from_qasm_file`

### What should happen?

It should be imported

### Any suggestions?

The int is larger than int64, which I assume is the root cause.

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

We fixed handling of giant integers in gate expression positions gh-12140, and this commit fixes the handling in conditionals. Unfortunately, this means pulling in big-int handling properly; the integers simply _are_ bigints, and we're not immediately converting them into something else.

The need to support this may influence how the Rust-space data models of `QuantumCircuit` evolve.

### Details and comments

Fix #12773

### CODE REVIEW

File: crates/qasm2/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ name = "qiskit_qasm2"
 doctest = false
 
 [dependencies]
+num-bigint = "0.4"
```
Comment (mtreinish) [inline crates/qasm2/Cargo.toml]: Should we move this to the workspace, because we use big int in accelerate too.

**Review by mtreinish — COMMENTED** [review]
This LGTM it's a pretty straightforward change, and you pre-empted my question about using `u128` as a short term bandaid with your unit tests. :) Just one inline comment about the cargo config, but that's small.

File: crates/qasm2/Cargo.toml (hunk 1 above)
Comment (jakelishman) [reply crates/qasm2/Cargo.toml]: Sure, done in 27e1689c6a.

**Review by mtreinish — APPROVED** [review]

**Comment by jakelishman** [comment]
@Mergifyio backport stable/1.1

**Pull request merged**